frontend/src/libs/packages/map/map.package.ts

```diff
@@ -155,26 +168,60 @@ class MapService implements IMapService {
     position: google.maps.LatLngLiteral,
     isOrigin = false,
     angle = 0,
-  ): void {
+  ): google.maps.Marker {
     this.throwIfMapNotInitialized();
 
-    const rotatedIconUrl = rotateImg(truckImg, angle);
-
-    new google.maps.Marker({
+    return new google.maps.Marker({
       position,
       map: this.map,
-      icon: isOrigin
-        ? {
-            url: rotatedIconUrl,
-            anchor: new google.maps.Point(
-              TRUCK_IMG_ANCHOR_X,
-              TRUCK_IMG_ANCHOR_Y,
-            ),
-            size: new google.maps.Size(TRUCK_IMG_WIDTH, TRUCK_IMG_HEIGHT),
-            scale: 1,
-          }
-        : undefined,
+      icon: createIcon(isOrigin, angle),
+    });
+  }
+
+  public async addRoute({
+    startPoint,
+    endPoint,
+  }: {
+    startPoint: google.maps.LatLngLiteral;
+    endPoint: google.maps.LatLngLiteral;
+  }): Promise<void> {
+    this.throwIfMapNotInitialized();
+
+    const path = await this.directionsService.route({
+      destination: endPoint,
+      origin: startPoint,
+      travelMode: google.maps.TravelMode.DRIVING,
     });
+
+    this.directionsRenderer.setOptions({
+      directions: path,
+      map: this.map,
+      preserveViewport: true,
+    });
+    await this.showInfoWindow(startPoint, endPoint);
+  }
+
+  public async showInfoWindow(
+    startpoit: google.maps.LatLngLiteral,
+    endPoint: google.maps.LatLngLiteral,
+  ): Promise<void> {
+    const anchor = this.addMarker(endPoint, false);
+
+    const startAddress = await this.getAddress(startpoit);
+    const endAddress = await this.getAddress(endPoint);
+
+    this.infoWindow.setContent(`${startAddress} → ${endAddress}`);
+    this.infoWindow.open({ map: this.map, anchor });
+  }
+
+  public async getAddress(place: google.maps.LatLngLiteral): Promise<string> {
+    this.throwIfMapNotInitialized();
+
+    const result: google.maps.GeocoderResponse = await this.geoCoder.geocode({
+      location: place,
+    });
+
+    return result.results[2].formatted_address;
```

Comment:
what means result.results[2] ?

Reply:
GeocoderResult looks like an array of address objects with different formats
![image](https://github.com/BinaryStudioAcademy/bsa-2023-towhub/assets/95139041/2ab187b8-314e-4a8e-8fb7-c91eff9f5932)